LIBOPING: fails to compile. Critical dependency of COLLECTD-MOD-PING. Related to NCURSES 6.2 to 6.3

Maintainer: @jow-  (Jo-Philipp Wich)
Environment: latest snapshot, config file attached below.

## Description:
When building the attached [.config](https://github.com/openwrt/packages/files/8310608/config.txt) file below, the build **always** fails at [packages/libs/liboping](https://github.com/openwrt/packages/tree/master/libs/liboping). This started to happen today, as this config has been building for several days until today March 19 2022. The same day of the new kernel upgrade from 5.10.104 to 5.10.107. Not sure if it is related to not. but I haven't found any other relevant change since the last successful build. This package is a dependency of `collectd-mod-ping`.

This is the console dump showing the error:
```make
make: Entering directory '/build/myopenwrt/R7800/feeds/packages/libs/liboping'
rm -f /build/myopenwrt/R7800/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/liboping-1.10.0/.built
touch /build/myopenwrt/R7800/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/liboping-1.10.0/.built_check
CFLAGS="-Os -pipe -g3 -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -mfloat-abi=hard -fmacro-prefix-map=/build/myopenwrt/R7800/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/liboping-1.10.0=liboping-1.10.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -DPIC -fpic  -I/build/myopenwrt/R7800/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-11.2.0_musl_eabi/usr/include -I/build/myopenwrt/R7800/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-11.2.0_musl_eabi/include/fortify -I/build/myopenwrt/R7800/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-11.2.0_musl_eabi/include " CXXFLAGS="-Os -pipe -g3 -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -mfloat-abi=hard -fmacro-prefix-map=/build/myopenwrt/R7800/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/liboping-1.10.0=liboping-1.10.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -DPIC -fpic  -I/build/myopenwrt/R7800/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-11.2.0_musl_eabi/usr/include -I/build/myopenwrt/R7800/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-11.2.0_musl_eabi/include/fortify -I/build/myopenwrt/R7800/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-11.2.0_musl_eabi/include " LDFLAGS="-L/build/myopenwrt/R7800/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-11.2.0_musl_eabi/usr/lib -L/build/myopenwrt/R7800/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-11.2.0_musl_eabi/lib -znow -zrelro " make  -C /build/myopenwrt/R7800/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/liboping-1.10.0/. AR="arm-openwrt-linux-muslgnueabi-gcc-ar" AS="arm-openwrt-linux-muslgnueabi-gcc -c -Os -pipe -g3 -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -mfloat-abi=hard -fmacro-prefix-map=/build/myopenwrt/R7800/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/liboping-1.10.0=liboping-1.10.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -DPIC -fpic" LD=arm-openwrt-linux-muslgnueabi-ld NM="arm-openwrt-linux-muslgnueabi-gcc-nm" CC="arm-openwrt-linux-muslgnueabi-gcc" GCC="arm-openwrt-linux-muslgnueabi-gcc" CXX="arm-openwrt-linux-muslgnueabi-g++" RANLIB="arm-openwrt-linux-muslgnueabi-gcc-ranlib" STRIP=arm-openwrt-linux-muslgnueabi-strip OBJCOPY=arm-openwrt-linux-muslgnueabi-objcopy OBJDUMP=arm-openwrt-linux-muslgnueabi-objdump SIZE=arm-openwrt-linux-muslgnueabi-size CROSS="arm-openwrt-linux-muslgnueabi-" ARCH="arm" ;
make[1]: Entering directory '/build/myopenwrt/R7800/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/liboping-1.10.0'
Making all in src
make[2]: Entering directory '/build/myopenwrt/R7800/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/liboping-1.10.0/src'
make  all-recursive
make[3]: Entering directory '/build/myopenwrt/R7800/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/liboping-1.10.0/src'
Making all in mans
make[4]: Entering directory '/build/myopenwrt/R7800/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/liboping-1.10.0/src/mans'
make[4]: Nothing to be done for 'all'.
make[4]: Leaving directory '/build/myopenwrt/R7800/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/liboping-1.10.0/src/mans'
make[4]: Entering directory '/build/myopenwrt/R7800/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/liboping-1.10.0/src'
/bin/bash ../libtool  --tag=CC   --mode=link arm-openwrt-linux-muslgnueabi-gcc -Wall -Os -pipe -g3 -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -mfloat-abi=hard -fmacro-prefix-map=/build/myopenwrt/R7800/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/liboping-1.10.0=liboping-1.10.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -DPIC -fpic   -version-info 3:0:3 -L/build/myopenwrt/R7800/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-11.2.0_musl_eabi/usr/lib -L/build/myopenwrt/R7800/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-11.2.0_musl_eabi/lib -znow -zrelro  -o liboping.la -rpath /usr/lib liboping_la-liboping.lo
OpenWrt-libtool: link: arm-openwrt-linux-muslgnueabi-gcc -shared  -fPIC -DPIC  .libs/liboping_la-liboping.o   -L/build/myopenwrt/R7800/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-11.2.0_musl_eabi/usr/lib -L/build/myopenwrt/R7800/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-11.2.0_musl_eabi/lib  -Os -mfloat-abi=hard -fstack-protector -Wl,-z -Wl,now -Wl,-z -Wl,relro   -Wl,-soname -Wl,liboping.so.0 -o .libs/liboping.so.0.3.0
OpenWrt-libtool: link: (cd ".libs" && rm -f "liboping.so.0" && ln -s "liboping.so.0.3.0" "liboping.so.0")
OpenWrt-libtool: link: (cd ".libs" && rm -f "liboping.so" && ln -s "liboping.so.0.3.0" "liboping.so")
OpenWrt-libtool: link: arm-openwrt-linux-muslgnueabi-gcc-ar cru .libs/liboping.a  liboping_la-liboping.o
/build/myopenwrt/R7800/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-11.2.0_musl_eabi/lib/gcc/arm-openwrt-linux-muslgnueabi/11.2.0/../../../../arm-openwrt-linux-muslgnueabi/bin/ar: `u' modifier ignored since `D' is the default (see `U')
OpenWrt-libtool: link: arm-openwrt-linux-muslgnueabi-gcc-ranlib .libs/liboping.a
OpenWrt-libtool: link: ( cd ".libs" && rm -f "liboping.la" && ln -s "../liboping.la" "liboping.la" )
/bin/bash ../libtool  --tag=CC   --mode=link arm-openwrt-linux-muslgnueabi-gcc -Wall -Os -pipe -g3 -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -mfloat-abi=hard -fmacro-prefix-map=/build/myopenwrt/R7800/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/liboping-1.10.0=liboping-1.10.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -DPIC -fpic   -L/build/myopenwrt/R7800/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-11.2.0_musl_eabi/usr/lib -L/build/myopenwrt/R7800/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-11.2.0_musl_eabi/lib -znow -zrelro  -o oping oping.o liboping.la -lm
OpenWrt-libtool: link: arm-openwrt-linux-muslgnueabi-gcc -Wall -Os -pipe -g3 -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -mfloat-abi=hard -fmacro-prefix-map=/build/myopenwrt/R7800/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/liboping-1.10.0=liboping-1.10.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z -Wl,now -Wl,-z -Wl,relro -DPIC -fpic -znow -zrelro -o .libs/oping oping.o  -L/build/myopenwrt/R7800/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-11.2.0_musl_eabi/usr/lib -L/build/myopenwrt/R7800/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-11.2.0_musl_eabi/lib ./.libs/liboping.so -lm
arm-openwrt-linux-muslgnueabi-gcc -DHAVE_CONFIG_H -I.   -DUSE_NCURSES=1  -I/build/myopenwrt/R7800/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-11.2.0_musl_eabi/usr/include -I/build/myopenwrt/R7800/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-11.2.0_musl_eabi/include/fortify -I/build/myopenwrt/R7800/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-11.2.0_musl_eabi/include  -Wall -Os -pipe -g3 -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -mfloat-abi=hard -fmacro-prefix-map=/build/myopenwrt/R7800/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/liboping-1.10.0=liboping-1.10.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -DPIC -fpic  -c -o noping-oping.o `test -f 'oping.c' || echo './'`oping.c
oping.c: In function 'update_graph_prettyping':
oping.c:1128:25: error: format not a string literal and no format arguments [-Werror=format-security]
 1128 |                         mvwprintw (ctx->window, /* y = */ 3, /* x = */ x + 2, symbol);
      |                         ^~~~~~~~~
oping.c: In function 'update_graph_histogram':
oping.c:1226:41: error: format not a string literal and no format arguments [-Werror=format-security]
 1226 |                                         hist_symbols_utf8[index]);
      |                                         ^~~~~~~~~~~~~~~~~
oping.c: In function 'update_host_hook':
oping.c:1601:38: warning: too many arguments for format [-Wformat-extra-args]
 1601 |                         HOST_PRINTF ("%zu bytes from %s (%s): icmp_seq=%u ttl=%i ",
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
oping.c:1578:45: note: in definition of macro 'HOST_PRINTF'
 1578 | # define HOST_PRINTF(...) wprintw(main_win, __VA_ARGS__)
      |                                             ^~~~~~~~~~~
cc1: some warnings being treated as errors
make[4]: *** [Makefile:595: noping-oping.o] Error 1
make[4]: Leaving directory '/build/myopenwrt/R7800/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/liboping-1.10.0/src'
make[3]: *** [Makefile:655: all-recursive] Error 1
make[3]: Leaving directory '/build/myopenwrt/R7800/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/liboping-1.10.0/src'
make[2]: *** [Makefile:430: all] Error 2
make[2]: Leaving directory '/build/myopenwrt/R7800/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/liboping-1.10.0/src'
make[1]: *** [Makefile:392: all-recursive] Error 1
make[1]: Leaving directory '/build/myopenwrt/R7800/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/liboping-1.10.0'
make: *** [Makefile:86: /build/myopenwrt/R7800/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/liboping-1.10.0/.built] Error 2
make: Leaving directory '/build/myopenwrt/R7800/feeds/packages/libs/liboping'
```
The failure happens 100% of the time and it is easy to reproduce. I have experienced it at least when building the same config file for the Netgear R7800 (arm) and Linksys E8450 (aarch64).

## STEPS TO REPRODUCE

1. Prepare an OpenWrt instance:
```bash
git clone -b master https://github.com/openwrt/openwrt.git openwrt
cd openwrt
./scripts/feeds update -a
./scripts/feeds install -a
```
2. copy the [.config](https://github.com/openwrt/packages/files/8310608/config.txt) file to the openwrt folder.
3. Try to build the image with:
```bash
make defconfig
make
```
To speedup the failure, this alternative process would bring the same issue faster:
```bash
make defconfig
make -j16 {toolchain,target}/compile
make -j1 V=sc feeds/packages/libs/liboping/compile
```

--- a/libs/liboping/Makefile
+++ b/libs/liboping/Makefile
@@ -9,7 +9,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=liboping
 PKG_VERSION:=1.10.0
-PKG_RELEASE:=2
+PKG_RELEASE:=$(AUTORELEASE)
 PKG_MAINTAINER:=Jo-Philipp Wich <jo@mein.io>
 PKG_LICENSE:=LGPL-2.1-or-later
 

--- a/libs/liboping/patches/05-fix-format-arguments-ncurses63.patch
+++ b/libs/liboping/patches/05-fix-format-arguments-ncurses63.patch
@@ -1,0 +1,30 @@
+--- a/src/oping.c
++++ b/src/oping.c
+@@ -1125,7 +1125,7 @@ static int update_graph_prettyping (ping
+ 			wattron (ctx->window, COLOR_PAIR(color));
+ 
+ 		if (has_utf8())
+-			mvwprintw (ctx->window, /* y = */ 3, /* x = */ x + 2, symbol);
++			mvwprintw (ctx->window, /* y = */ 3, /* x = */ x + 2, "%s", symbol);
+ 		else
+ 			mvwaddch (ctx->window, /* y = */ 3, /* x = */ x + 2, symbolc);
+ 
+@@ -1223,7 +1223,7 @@ static int update_graph_histogram (ping_
+ 			mvwaddch (ctx->window, /* y = */ 3, /* x = */ x + 2, ' ');
+ 		else if (has_utf8 ())
+ 			mvwprintw (ctx->window, /* y = */ 3, /* x = */ x + 2,
+-					hist_symbols_utf8[index]);
++					"%s", hist_symbols_utf8[index]);
+ 		else
+ 			mvwaddch (ctx->window, /* y = */ 3, /* x = */ x + 2,
+ 					hist_symbols_acs[index] | A_ALTCHARSET);
+@@ -1600,8 +1600,7 @@ static void update_host_hook (pingobj_it
+ 
+ 			HOST_PRINTF ("%zu bytes from %s (%s): icmp_seq=%u ttl=%i ",
+ 					data_len, context->host, context->addr,
+-					sequence, recv_ttl,
+-					format_qos (recv_qos, recv_qos_str, sizeof (recv_qos_str)));
++					sequence, recv_ttl);
+ 			if ((recv_qos != 0) || (opt_send_qos != 0))
+ 			{
+ 				HOST_PRINTF ("qos=%s ",

